test/exe/main_common_test.cc

```diff
@@ -25,14 +25,25 @@
 
 namespace Envoy {
 
-TEST(MainCommon, ConstructDestruct) {
+TEST(MainCommon, ConstructDestructHotRestartEnabled) {
   if (!Envoy::TestEnvironment::shouldRunTestForIpVersion(Network::Address::IpVersion::v4)) {
     return;
   }
   std::string config_file = Envoy::TestEnvironment::getCheckedEnvVar("TEST_RUNDIR") +
                             "/test/config/integration/google_com_proxy_port_0.v2.yaml";
-  const char* argv[] = {"envoy-static", "-c", config_file.c_str(), nullptr};
-  MainCommon main_common(3, const_cast<char**>(argv), false);
+  const char* argv[] = {"envoy-static", "-c", config_file.c_str(), "--base-id", "1", nullptr};
+  MainCommon main_common(ARRAY_SIZE(argv) - 1, const_cast<char**>(argv));
+}
+
+TEST(MainCommon, ConstructDestructHotRestartDisabled) {
+  if (!Envoy::TestEnvironment::shouldRunTestForIpVersion(Network::Address::IpVersion::v4)) {
+    return;
+  }
+  std::string config_file = Envoy::TestEnvironment::getCheckedEnvVar("TEST_RUNDIR") +
+                            "/test/config/integration/google_com_proxy_port_0.v2.yaml";
+  const char* argv[] = {"envoy-static",          "-c",   config_file.c_str(), "--base-id", "2",
+                        "--disable-hot-restart", nullptr};
+  MainCommon main_common(ARRAY_SIZE(argv) - 1, const_cast<char**>(argv));
```

Comment:
nit/curious  - would wrapping these in EXPECT_NO_THROW make the test clearer that it is expected that these construct with no issue? (Context for my question - I was doing a quick look for EXPECT_... and got lost for a few seconds. :) )

Reply:
seems reasonable.  I guess if it threw otherwise it would stop all the tests in the file?